pos_stock_available_online/models/stock_notifier_pos_mixin.py

```diff
@@ -0,0 +1,54 @@
+from odoo import models
+
+
+class StockNotifierPosMixin(models.AbstractModel):
+    _name = "stock.notifier.pos.mixin"
+    _description = "Stock Notifier POS Mixin"
+
+    def _prepare_pos_message(self, warehouse=None):
```

Comment:
The `_prepare_pos_message` is called directly in the warehouse loop in the `_notify_pos` method, which gets the list of warehouses using the `_get_warehouses_to_notify` method in this case warehouse will always be in the argument values, so it can be left as a mandatory argument.

```python
def _prepare_pos_message(self, warehouse):
     """
     Return prepared message to send to POS
     """
     self.ensure_one()
     return warehouse._prepare_vals_for_pos(self.product_id)
```